src/with-render.js

```diff
@@ -10,7 +10,7 @@ function attachShadow (elem) {
 
 export const withRender = (Base = HTMLElement) => class extends Base {
   propsChangedCallback () {
-    this[_shadowRoot] = this[_shadowRoot] || (this[_shadowRoot] = attachShadow(this));
+    this[_shadowRoot] = this[_shadowRoot] || (this[_shadowRoot] = (this.shadowRoot || attachShadow(this)));
```

Comment:
Sure, it will fail for closed shadows with or without this change. But since 99% of shadows are open, why not fix it for those?